Packs/GsuiteAuditor/Integrations/GsuiteAuditor/GsuiteAuditor.py

```diff
@@ -0,0 +1,344 @@
+
+    :param client: client object which is used to get response from api
+    :param args: command arguments.
+
+    :return: CommandResults object with context and human-readable.
+    """
+    arguments = prepare_args_for_activities_list(args)
+    response = client.http_request(
+        url_suffix=URL_SUFFIX.format(args.get('user_key', 'all'), args.get('application_name')),
+        params=arguments)
+
+    # Readable Output
+    readable_items = prepare_readable_items(response.get('items', []))
+
+    readable_output = f'### Next Page Token: {response["nextPageToken"]}\n' if response.get("nextPageToken") else ''
+    readable_output += tableToMarkdown(
+        'Total Retrieved {}: {}'.format('Activities', len(response.get('items', []))),
+        readable_items,
+        ['time', 'applicationName', 'email', 'profileId', 'ipAddress', 'events']) if response.get(
+        'items') else 'No activities found for the given ' \
+                      'argument(s). '
+    # Context
+    context_outputs = prepare_output_for_activities_list(response)
+
+    return CommandResults(
+        outputs=context_outputs,
+        readable_output=readable_output,
+        raw_response=response
+    )
+
+
+@logger
+def fetch_incidents(client: GSuiteClient, first_fetch_time: str, fetch_limit: int, applications: Any) -> List[Dict]:
+    total_items = []
+    last_run_dict = demisto.getLastRun()
+    last_run = last_run_dict.get('last_run')
+    last_ids = last_run_dict.get('last_ids', [])
+    last_leftover_items = last_run_dict.get('last_leftover_items', [])
+    new_leftover_items = []
+    if not last_run:  # this is the first run
+        last_run = dateparser.parse(first_fetch_time).strftime(DATE_FORMAT)
+
+    end_time = create_end_time(last_run, "1 hours")
+
+    if len(last_leftover_items) != 0:
+        total_items = last_leftover_items[:fetch_limit]
+        new_leftover_items = last_leftover_items[fetch_limit:]
+
+    if len(total_items) < fetch_limit:
+        for app in applications:
+            total_items.extend(fetch_request(client, app, last_run, end_time))
+
+    sorted_items = sorted(total_items, key=lambda k: k['id']['time'])  # sort the data from earlist to last.
+
+    incidents = []
+    counter = 0
+    for i in range(0, len(sorted_items)):
+        item = sorted_items[i]
+        if counter == fetch_limit:
+            new_leftover_items.extend(sorted_items[i:])
+            break
+
+        if item['id']['uniqueQualifier'] in last_ids:
+            continue
+
+        counter += 1
+        incident = {
+            'name': f"GSuite Auditor event {item['id']['applicationName']} {item['id']['uniqueQualifier']}",
+            'occurred': item['id']['time'],
+            'rawJSON': json.dumps(item)
+        }
+        incidents.append(incident)
+
+    if len(incidents) > 0:
+        new_last_run = incidents[-1]['occurred']
+    elif end_time < time.ctime():
+        new_last_run = end_time
+    else:
+        new_last_run = last_run
```

Comment:
so if I get this:
if we got results - the next run starts from the last incident.

if we don't have results and this run's end time was earlier than the current time, the next start will run from the current end time. <- why is this correct?

if non of the above - repeat the current run.

Reply:
in case we are still retrieving results from the past , and we made a run with no results, we don't want to repeat this run. 
for example my first run is from 10:00 and now it's 13:00 , I ran from 10:00 to 11:00 (we created a 1 hour limit) and no results were fetched, we wand to set the new end time to 11 or else we'll get stuck. if we are in 'real time' run we want to stay foot until the 1 hour frame passes. 
hope this makes sense I added a dev comment to make it clearer